src/ls/definition/enum_id.cr

```diff
@@ -2,20 +2,35 @@ module Mint
   module LS
     class Definition < LSP::RequestMessage
       def definition(node : Ast::EnumId, server : Server, workspace : Workspace, stack : Array(Ast::Node))
-        return unless name = node.name
-        return unless enum_node =
-                        workspace.ast.enums.find(&.name.value.==(name.value))
+        name = node.name
 
-        return if Core.ast.enums.includes?(enum_node)
+        # When `.name` is nil the node is used as a CONSTANT
+        if name.nil?
+          stack.each do |parent|
+            case parent
+            when Ast::Component, Ast::Store
```

Comment:
It can also be `Ast::Suite`, `Ast::Module` and `Ast::Provider`.

Reply:
> I had a quick look and it seems they aren't being formatted (so are removed when you format the file) and also not type checked properly. Can look into fixing that in a separate PR?

Nice find :+1: yeah go ahead, should be a quick fix.

Reply:
Updated to support `Ast::Provider` now too :+1: 

Haven't had much chance to test this PR in earnest yet (unfortunately have to use Typescript in work :sob:) but overall should work OK. Let me know if you find any issues!